packages/@storefront/core/src/services/url.ts

```diff
@@ -157,15 +157,40 @@ class UrlService extends BaseService<UrlService.Options> {
 
   filterState(state: Store.State) {
     const { session: { config, ...session }, data, ...rootConfig } = state;
-    const products = this.app.config.history.length === 0 ? [] : data.present.products;
+    let { navigations, products, template } = data.present;
+    let {
+      navigations: autocompleteNavigations,
+      products: autocompleteProducts,
+      template: autocompleteTemplate,
+    } = data.present.autocomplete;
+
+    if (this.app.config.history.length === 0) {
+      autocompleteNavigations = [];
+      autocompleteProducts = [];
+      autocompleteTemplate = <any>{};
+      navigations = { allIds: [], byId: {}, sort: [] };
+      products = [];
+      template = <any>{};
```

Comment:
Will TypeScript complain if you remove the cast?

Reply:
Yes, since original assignment is based on the state object, TypeScript expects the `template` properties to exist. Resolving.